core/src/components/menu/menu.tsx

```diff
@@ -322,23 +322,9 @@ export class Menu implements ComponentInterface, MenuI {
     }
   }
 
-  @Listen('click', { capture: true })
-  onBackdropClick(ev: any) {
-    // TODO(FW-2832): type (CustomEvent triggers errors which should be sorted)
-    if (this._isOpen && this.lastOnEnd < ev.timeStamp - 100) {
-      const shouldClose = ev.composedPath ? !ev.composedPath().includes(this.menuInnerEl) : false;
-
-      if (shouldClose) {
-        ev.preventDefault();
-        ev.stopPropagation();
-        this.close();
-      }
-    }
-  }
-
   onKeydown(ev: KeyboardEvent) {
     if (ev.key === 'Escape') {
-      this.close();
+      this.close(undefined, ESCAPE);
```

Comment:
This is different from how the ion-modal works. When the modal is closed using the escape key, it emits a role of 'backdrop'. Should we match the modal behavior here?

Reply:
Changed it to match the other overlays for consistency: https://github.com/ionic-team/ionic-framework/pull/29954/commits/721bc45f9562acb32b8c9c5fbf156995ab480ca8